vendor/engines/saml_authentication/spec/controllers/saml_authentication/sessions_controller_spec.rb

```diff
@@ -135,4 +140,66 @@
       end
     end
   end
+
+  describe "#destroy" do
+    let(:user) { create(:user) }
+    before do
+      request.env["devise.mapping"] = Devise.mappings[:user]
+      sign_in user
+    end
+
+    it "logs the user out logout" do
+      delete :destroy
+      expect(controller.current_user).to be_blank
+    end
+
+    it "redirects to the slo path" do
+      delete :destroy
+      expect(response.location).to start_with(idp_slo_path)
+    end
+
+    it "includes the username in the logout request" do
+      delete :destroy
+      uri = URI.parse(response.location)
+      deflated = Base64.decode64 Rack::Utils.parse_query(uri.query)["SAMLRequest"]
+      xml = Zlib::Inflate.new(-Zlib::MAX_WBITS).inflate(deflated)
+      doc = Nokogiri::XML(xml)
+      expect(doc.css("saml|NameID").first.text).to eq(user.username)
+    end
+  end
+
+  describe "#idp_sign_out" do
+    let(:user) { create(:user) }
+    before do
+      request.env["devise.mapping"] = Devise.mappings[:user]
+      request.host = "localhost:3000" # without this, we're getting redirected to http://test.host/
```

Comment:
I could not figure out what was going on here, and eventually gave up. I was expecting it to work with what we have defined in `Rails.application.routes.default_url_options` in `environments/test.rb`

Reply:
FWIW, this is because `ActionDispatch::TestRequest` [defaults `HTTP_HOST` to `test.host`](https://github.com/rails/rails/blob/e53f45c57263e4e996bff3d4ba26342cadd3cca5/actionpack/lib/action_dispatch/testing/test_request.rb#L7). As a result, when a request like this is processed by Rails in a functional test, it uses that request’s host for any URL generation, because `default_url_options` kicks in only when the host is unknown (like in mailer contexts).

Reply:
What I was getting confused by is we have other controller specs that don't have this issue, and we don't specifically test `request.host`.